plugins/modules/packaging/os/pacman.py

```diff
@@ -271,6 +271,10 @@ def run(self):
             if not (self.m.params["name"] or self.m.params["upgrade"]):
                 self.success()
 
+        # Avoid shadowing lack of changes in the following stages
+        # so that update_cache: yes doesn't always return changed
+        self.changed = False
```

Comment:
I don't think this is correct. Instead `update_package_db()` should be changed to correctly determine whether the package database was updated or not. If that's not possible, it has to always be treated as changed.

Reply:
pacman's manual seem to indicate that using a single `--refresh` should not refresh databases if they appear up to date, but unless I'm doing something weird here, it always refresh all databases.  meaning that it seems to always change anyway?

Reply:
`pacman -Sy` doesn't refresh databases that weren't updated. So if you run that twice in a row, the second invocation should have no change (except if you were unlucky and the DB on the server actually changed between the two calls :) ).

It only shows that when the output of `pacman -Sy` is in a proper terminal though. If you for exmaple pipe it through `cat`, you cannot distinguish between the two cases.

`pacman --sync --refresh --debug 2>&1 | grep ".db: response code"` does allow you to do that, though (see my comment below).

Reply:
nevermind that, it's my local caching server (flexo)  that seem to being doing something weird.